data_pipelines_cli/dbt_cloud_api_client.py

```diff
@@ -0,0 +1,224 @@
+import json
+import requests
+
+
+class DbtCloudApiClient:
+    """A class used to create dbt Cloud project using API v3"""
+
+    def __init__(self, host_url, account_id, token):
+        self.host_url = host_url
+        """Base URL differs for Multi and Single-Tenant Deployments"""
+
+        self.account_id = account_id
+        """
+        To find your user ID in dbt Cloud, read the following steps:
+        1. Go to Account Settings, Team, and then Users,
+        2. Select your user,
+        3. In the address bar, the number after /users is your user ID.
+        """
+
+        self.token = token
+        """You can find your User API token in the Profile page under the API Access label"""
+
+    def request(self, url, data):
+        headers = {
+            "Accept": "application/json",
+            "Content-Type": "application/json",
+            "Authorization": f"Token {self.token}"
+        }
+        response = requests.post(
+            url=url,
+            data=data,
+            headers=headers
+        )
+        res = json.loads(response.content)
+        if not res["status"]["is_success"]:
+            raise Exception(res["status"]["user_message"] + "\n" + res["data"])
+        return res
+
+    def create_project(self, name):
+        """
+        Note: the dbt_project_subdirectory is an optional field which allows you to point
+        dbt Cloud to a subdirectory that lives within the root folder of your target repository.
+        This means dbt Cloud will look for a dbt_project.yml file at that location.
+
+        :param name: Name of the project
+        :return: ID of created project
+        """
+        new_project = {
+            "id": None,
+            "account_id": self.account_id,
+            "name": name,
+            "dbt_project_subdirectory": None,
+            "connection_id": None,
+            "repository_id": None
```

Comment:
Maybe you could first create repo and then just assign it here instead doing separate call?